pypeit/wavetilts.py

```diff
@@ -188,34 +188,45 @@ def show(self, waveimg=None, wcs_match=True, in_ginga=True, show_traces=False):
                 If True, show the traces of the tilts on the image.
 
         """
+        # get tilt_img_dict
+        if (Path(self.calib_dir).resolve() / self.tiltimg_filename).exists():
+            tilt_img_dict = buildimage.TiltImage.from_file(Path(self.calib_dir).resolve() / self.tiltimg_filename)
+        else:
+            msgs.error(f'Tilt image {str((Path(self.calib_dir).resolve() / self.tiltimg_filename))} NOT FOUND.')
+
         # get slits
+        slitmask = None
         if (Path(self.calib_dir).resolve() / self.slits_filename).exists():
             slits = slittrace.SlitTraceSet.from_file(Path(self.calib_dir).resolve() / self.slits_filename)
-            slitmask = slits.slit_img(initial=True, flexure=self.spat_flexure)
-            left, right, mask = slits.select_edges(flexure=self.spat_flexure)
-            gpm = mask == 0
+            _slitmask = slits.slit_img(initial=True, flexure=self.spat_flexure)
+            if tilt_img_dict.image.shape != _slitmask.shape:
+                msgs.warn('Tilt image and other calibration frames have different shapes. '
+                          'Cannot load slits nor Wave image.')
+                # resize slitmask to match tilt_img_dict.image.shape
+                slitmask = arc.resize_mask2arc(tilt_img_dict.image.shape, _slitmask)
+                # but, still, we cannot really use slits, so we set it to None
+                slits = None
+            else:
```

Comment:
There is a resizing routine for the slits as well. Again, you should not be throwing any warnings here because the slits being a different shape than the arc/tilt image is a common occurrence for X-shooter.  See the __init__.py method in wavetilts.py. There is a method called arc.resize_slits2arc.  

What you should be doing here is getting rid of the if/else because the routines are designed to work no matter what the shapes are as long as they are integer multiples of each other. 

_slitmask = slits.slit_img(initial=True, flexure=self.spat_flexure)
slitmask = arc.resize_mask2arc(tilt_img_dict.image.shape, _slitmask)
_left, _right, _mask = slits.select_edges(flexure=self.spat_flexure)
left = arc.resize_slits2arc(tilt_img_dict.image.shape, _slitmask.shape, _left)
right = left = arc.resize_slits2arc(tilt_img_dict.image.shape, _slitmask.shape, _right)

Then I'm a bit unclear on what you are doing with the mask, but you can I think use the arc.resize_mask2arc to reshape that as well. No if/then is needed and no warnings should be thrown. The resizing routines return the input image if the shapes are the same.